src/main/java/org/apache/xml/security/stax/ext/XMLSecurityConstants.java

```diff
@@ -73,17 +74,34 @@ protected XMLSecurityConstants() {
     }
 
     /**
-     * Generate bytes of the given length using the supplied algorithm in RANDOM_ALGORITHM_KEY or,
-     * if not specified, use SecureRandom instance from default constructor. The SecureRandom
+     * Generate bytes of the given length using the algorithm supplied in {@value #RANDOM_ALGORITHM_KEY} or,
+     * if not specified, use a {@code SecureRandom} instance from default constructor. The {@code SecureRandom}
      * instance that backs this method is cached for efficiency.
      *
-     * @return a byte array of the given length
+     * @return a new byte array of the given length
      * @throws XMLSecurityException
      */
     public static byte[] generateBytes(int length) throws XMLSecurityException {
+
+        SecureRandom rnd = SECURE_RANDOM;
+        if (rnd == null) {
+            synchronized (SECURE_RANDOM_LOCK) {
+                rnd = SECURE_RANDOM;
+                if (rnd == null) {
+                    try {
+                        String prngAlgorithm = System.getProperty("org.apache.xml.security.securerandom.algorithm");
```

Comment:
Not sure if this code is intended to run under a Security Manager, but if so, it should be wrapped inside doPrivileged. @coheigea ?

Reply:
Yes please wrap it in doPrivileged @ppalaga like for https://github.com/apache/santuario-xml-security-java/blob/fa63d6fb5acb0361c491c2c4054ce38eb6907746/src/main/java/org/apache/xml/security/encryption/XMLCipherUtil.java#L37